### PR TITLE
IBX-7894: Added missing languages for DeepL and bumped API version to v2

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| :ticket: Issue | IBX-7894 |
|----------------|-----------|

#### Description:
Added missing languages for DeepL because our list was very old and some code of languages are deprecated. Also, bumped the API version to v2 as version v1 is no longer supported.

The `source_lang` accepts only two letters of language code

### CODE REVIEW

**Comment by adamwojs** [comment]
@mateuszdebinski Could you please provide some minimal PR description?

File: src/lib/Client/Deepl.php [hunk 1]
```diff
@@ -82,13 +91,32 @@ private function normalized(string $languageCode): string
...
+            return 'EN-US';
+        }
+
+        if ('pt_BR' === $languageCode) {
+            return 'PT-BR';
+        }
+
+        if ('PT' === $code) {
+            return 'PT-PT';
+        }
```
Comment (adamwojs) [inline src/lib/Client/Deepl.php:116]: Can we introduce here some conversion map? It should be even parameter injected into service ?

**Review by adamwojs — APPROVED** [review]
Thank you @mateuszdebinski

File: src/lib/Client/Deepl.php [hunk 2]
```diff
@@ -15,8 +15,30 @@
 
 class Deepl implements ClientInterface
 {
+    /**
+     * List of available code https://developers.deepl.com/docs/resources/supported-languages.
```
Comment (konradoboza) [inline src/lib/Client/Deepl.php:19]:
```suggestion
     * List of available codes https://developers.deepl.com/docs/resources/supported-languages.
```

**Review by konradoboza — APPROVED** [review]

**Comment by bogusez** [comment]
@mateuszdebinski after applying patch I have an issue 

Screenshot 2024-10-02 at 13 36 09

**Comment by bogusez** [comment]
@mateuszdebinski there is also an issue with missing service on Headless edition. Automated translation feature not working: https://issues.ibexa.co/browse/IBX-8992

**Comment by mateuszdebinski** [comment]
I added a change that eliminates the bug for Headless. As for the first bug with DeepL, we do not support the free version, if this were to change it would probably be a new ticket.

File: src/lib/Client/Deepl.php [hunk 3]
```diff
@@ -15,8 +15,30 @@
...
+        'AR', 'BG', 'CS', 'DA', 'DE', 'EL', 'EN-GB', 'EN-US', 'ES', 'ET', 'FI', 'FR',
+        'HU', 'ID', 'IT', 'JA', 'KO', 'LT', 'LV', 'NB', 'NL', 'PL', 'PT-BR', 'PT-PT', 'RO',
+        'RU', 'SK', 'SL', 'SV', 'TR', 'UK', 'ZH-HANS', 'ZH-HANT',
+    ];
+
     private string $authKey;
 
+    /**
+     * @param array<string, string> $languageMap
+     */
```
Comment (konradoboza) [inline src/lib/Client/Deepl.php:31]:
```suggestion
    /** @var array<string, string> $languageMap */
```

**Review by konradoboza — APPROVED** [review]

File: src/bundle/IbexaAutomatedTranslationBundle.php [hunk 4]
```diff
@@ -21,6 +23,11 @@ public function getParent(): ?string
     public function build(ContainerBuilder $container): void
     {
         parent::build($container);
+        $loader = new YamlFileLoader($container, new FileLocator(__DIR__ . '/Resources/config'));
+
+        if ($container->hasExtension('ibexa_fieldtype_page')) {
```
Comment (ViniTou) [inline src/bundle/IbexaAutomatedTranslationBundle.php:28]: Prolly minor, as those two should be one package, but shouldn't it be `ibexa_page_builder`?

// and `page_builder_services.yaml`

Comment (mateuszdebinski) [reply src/bundle/IbexaAutomatedTranslationBundle.php:28]: On the one hand, we have PageBuilderFieldEncoder and on the other hand, all dependencies are to `FieldTypePage`, that's why I gave that the required extension is `ibexa_fieldtype_page`

**Review by adamwojs — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Pull request merged**